src/files/manifest/readVariantManifests.ts

```diff
@@ -0,0 +1,60 @@
+import { defaultManifestFileName, defaultVariantsDir } from "../../params.js";
+import { VariantManifestMap } from "./types.js";
+import path from "path";
+import fs from "fs";
+import { readManifest } from "./readManifest.js";
+
+export function readVariantManifests({
+    dir,
+    variantsDir = defaultVariantsDir,
+    variants,
+    manifestFileName = defaultManifestFileName
+}: {
+    dir: string;
+    variantsDir: string;
+    variants?: string[];
+    manifestFileName?: string;
+}): VariantManifestMap {
+    const variantsMap: VariantManifestMap = {};
+    const fullVariantsDirPath = path.join(dir, variantsDir);
```

Comment:
should `dir` and `variantsDir` be somewhat sanitized so that `fullVariantsDirPath` is correctly built? It is crucial that it is a correct path

Reply:
Actually if the path is empty or it does not exist it will either throw an error (caught in the parent function) or return an empty array so there will be no variants to modify. I'd say this is enough